tests/transparent.rs

```diff
@@ -0,0 +1,97 @@
+use snafu::prelude::*;
+use std::error::Error;
+
+#[derive(Debug, Snafu)]
+enum AlphaError {
+    AlphaDummy,
+}
+
+#[derive(Debug, Snafu)]
+enum BetaError {
+    BetaDummy { source: AlphaError },
+}
+
+#[derive(Debug, Snafu)]
+enum ErrorTest {
+    #[snafu(transparent)]
+    Alpha { source: AlphaError },
+
+    #[snafu(transparent)]
+    Beta { source: BetaError },
+}
+
+fn alpha() -> Result<i32, AlphaError> {
+    AlphaDummySnafu.fail()
+}
+
+fn beta() -> Result<i32, BetaError> {
+    let a = alpha().context(BetaDummySnafu)?;
+    Ok(a + 2)
+}
+
+fn example() -> Result<i32, ErrorTest> {
+    let a = 2;
+    let b = beta()?;
+    Ok(a * 10 + b)
+}
+
+fn check<T: std::error::Error>() {}
+
+#[test]
+fn implements_error() {
+    check::<ErrorTest>();
+
+    let error = example().unwrap_err();
+    // skpping `ErrorTest`, directly going yo BetaError's source
+    assert_eq!(error.source().unwrap().to_string(), "AlphaDummy");
+}
+
+#[test]
+fn implements_display() {
+    let error = example().unwrap_err();
+    // no `Beta: ` prefix
+    assert_eq!(error.to_string(), "BetaDummy: AlphaDummy");
+}
+
+mod with_backtraces {
```

Comment:
Did this module (and the next) come about because of some specific issue during implementation, or just as insurance?

Reply:
Neither, just used no_context.rs as base: https://github.com/shepmaster/snafu/blob/19f667a80a0d2b5474ac1dc1b852f7cbea96d1ec/tests/no_context.rs#L45-L86

Reply:
Just realized why I chose that particular one - the rules for when `snafu(transparent)` is allowed are the same as `snafu(context(false))`, since it is implied by `transparent`.